### PR TITLE
feature(logging): Ensure logged messages are formatted correctly

### PR DESCRIPTION
**Pull request by Cawllec**
- Ensures logged messages have the correct bugsnag formatting
- Ensures custom loggers receive the formatted message rather than the raw message
- Adds a new public method - 'format_message'

### CODE REVIEW

File: lib/bugsnag/configuration.rb [hunk 1]
```diff
@@ -88,9 +88,6 @@ def initialize
       # Set up logging
       self.logger = Logger.new(STDOUT)
       self.logger.level = Logger::INFO
-      self.logger.formatter = proc do |severity, datetime, progname, msg|
-        "** [Bugsnag] #{datetime}: #{msg}\n"
```
Comment (kattrali) [inline lib/bugsnag/configuration.rb]: Keep the formatter as before, only removing the `[Bugsnag]` component

File: lib/bugsnag/configuration.rb [hunk 2]
```diff
@@ -169,19 +166,26 @@ def clear_request_data
     ##
     # Logs an info level message
     def info(message)
-      logger.info(message)
+      logger.info(format_message(message))
```
Comment (kattrali) [inline lib/bugsnag/configuration.rb]: Use the [block form of Logger logging methods](https://ruby-doc.org/stdlib-1.9.2/libdoc/logger/rdoc/Logger.html#method-i-info) to specify the program name which sent the message. This differentiates Bugsnag log messages from others sent using the same logger.

```diff
- logger.info(format_message(message))
+ logger.info('Bugsnag') { message }
```

File: lib/bugsnag/configuration.rb [hunk 3]
```diff
@@ -169,19 +166,26 @@ def clear_request_data
...
     ##
     # Logs a debug level message
     def debug(message)
-      logger.debug(message)
+      logger.debug(format_message(message))
+    end
+
+    ##
+    # Formats a message being logged by Bugsnag
+    def format_message(message)
```
Comment (kattrali) [inline lib/bugsnag/configuration.rb]: While this is unneeded in a simpler implementation, it should be noted that this method should have been private as it has no role in the public interface.

**Comment by Cawllec** [comment]
I've adjusted how this works based on your feedback - the formatters back in place and the `PROG_NAME` const will be delivered as the program name in all logger cases.

**Comment by kattrali** [comment]
Looks good, I made one adjustment to make the `PROG_NAME` constant private and title-case it as before.

**Pull request merged**